src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -140,36 +271,54 @@ public RunVersionOne CreateRun(Run v2Run)
                 }
                 else
                 {
-                    run = new RunVersionOne()
-                    {
-                        Architecture = v2Run.Architecture,
-                        AutomationId = v2Run.AutomationLogicalId,
-                        BaselineId = v2Run.BaselineInstanceGuid,
-                        Files = v2Run.Files?.ToDictionary(v => v.Key, v => CreateFileDataVersionOne(v.Value)),
-                        Id = v2Run.InstanceGuid,
-                        LogicalLocations = v2Run.LogicalLocations?.ToDictionary(v => v.Key, v => CreateLogicalLocationVersionOne(v.Value)),
-                        Properties = v2Run.Properties,
-                        Results = new List<ResultVersionOne>(),
-                        Rules = v2Run.Resources?.Rules?.ToDictionary(v => v.Key, v => CreateRule(v.Value)),
-                        StableId = v2Run.LogicalId,
-                        Tool = CreateTool(v2Run.Tool)
-                    };
+                    _currentV2Run = v2Run;
 
+                    // We need to create the run before we start working on children
+                    // because some of them will need to refer to _currentRun
+                    run = new RunVersionOne();
                     _currentRun = run;
 
+                    run.Architecture = v2Run.Architecture;
+                    run.AutomationId = v2Run.AutomationLogicalId;
+                    run.BaselineId = v2Run.BaselineInstanceGuid;
+                    run.Files = v2Run.Files?.ToDictionary(v => v.Key, v => CreateFileData(v.Value));
+                    run.Id = v2Run.InstanceGuid;
+                    run.Invocation = CreateInvocation(v2Run.Invocations?[0]);
+                    run.LogicalLocations = v2Run.LogicalLocations?.ToDictionary(v => v.Key, v => CreateLogicalLocation(v.Value));
+                    run.Properties = v2Run.Properties;
+                    run.Results = new List<ResultVersionOne>();
+                    run.Rules = v2Run.Resources?.Rules?.ToDictionary(v => v.Key, v => CreateRule(v.Value));
+                    run.StableId = v2Run.LogicalId;
+                    run.Tool = CreateTool(v2Run.Tool);
+
+                    RemoveResponseFilesFromFileDictionary();
+
                     // Stash the entire v2 run in this v1 run's property bag
-                    run.SetProperty($"{FromPropertyBagPrefix}/run", JsonConvert.SerializeObject(v2Run, Utilities.JsonSettingsV2));
+                    run.SetProperty($"{FromPropertyBagPrefix}/run", JsonConvert.SerializeObject(v2Run, Utilities.JsonSettingsV2Compact));
                 }
             }
 
             return run;
         }
 
-        public ToolVersionOne CreateTool(Tool v2Tool)
+        internal void RemoveResponseFilesFromFileDictionary()
+        {
+            if (_currentRun.Invocation?.ResponseFiles != null)
```

Comment:
This is one of those things that is almost always ok, but could conceivably not be. There's no harm in leaving the file dictionary entries for the response files, in case it's needed for some other reason.

Reply:
OK. I did this because they are not there going the other direction.

Reply:
Removed